eclipse-theia/theia
# Show file changes as a tree in Source Control view

Replaces file change list in Source Control view with tree view.  This fixes https://github.com/eclipse-theia/theia/issues/4835

#### What it does
The change lists are replaced by a single tree that uses TreeWidget.  A toggle button has been added so the user can toggle between a flat view and a tree view.

#### What will not be covered by this PR
The diff widget

#### How to test
Try out the Source Control view.  This is the only view that should have changed.  The 'flat' mode should look mostly the same but even in flat mode it is now in a tree, with each group (staged, unstaged etc) being collapsable.  Test with both the Git vs-code built-in and with the native Git extension.

#### Review checklist

- [x ] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
